Add basic Storage testing

Kafcis opened the pull request:
Changed TestSetup index to follow normal operation
Caught a bug in TaskList Class from testing and fixed it

Kafcis committed:
Add basic Storage testing
Changed TestSetup index to follow normal operation
Caught a bug in TaskList Class from testing and fixed it

Kafcis committed:
Checkstyle fix

Kafcis committed:
Changed test names to fit conventions

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,72 @@
+        storage.saveTaskListData(taskList);
+        assertTrue(areFilesSame("src/test/java/athena/loadTask.csv", "src/test/java/athena/TaskAnswer.csv"));
+
+
```
Unnecessary empty lines

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,72 @@
+        try {
+            answer = new BufferedReader(new FileReader(file1));
+
+
```
Unnecessary empty line

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,72 @@
+            return false;
+        }
+
+
```
Empty lines

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,72 @@
+            return true;
+        }
+
+
```
Empty lines

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,72 @@
+        assertTrue(areTaskListsSame(tester, taskList));
+    }
+
+    private boolean areTaskListsSame(TaskList tester, TaskList other) {
+        if (tester.getMaxIndex() != (other.getMaxIndex())) {
+            return false;
+        } else if (tester.getTasks().equals(other.getTasks())) {
+            return false;
+        } else {
+            return true;
+        }
+
+
+    }
```
It would be better override `equals` in `TaskList` instead

https://github.com/AY2021S1-CS2113T-W12-2/tp/blob/30bf904954319f483b762ee3c7f5a378e494c4b7/src/main/java/athena/TaskList.java#L179-L189

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,72 @@
+
+
+    }
+
+
```
Empty lines

daniellimws reviewed (changes requested):
Looks good. Just a comment about using `equals` to compare two `TaskList`s.

Kafcis committed:
Update StorageTest
Included equals overide in TaskList, need to change maxIndex to maxNumber after #61
Removed unecessary blank lines

Kafcis committed:
Add save.csv
Prevent merge conflict

Kafcis committed:
Merge branch 'master' into 52-tests-for-storage

Kafcis committed:
Remove blank lines

Kafcis committed:
Update StorageTest.java

Kafcis committed:
Update StorageTest
Changed to fit to #61

Kafcis committed:
Update StorageTest

Kafcis committed:
Checkstyle fix

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,53 @@
+        TaskList taskList = TestSetup.getTestTaskList();
+        Storage storage = new Storage("src/test/java/athena/loadTask.csv", ui);
+        storage.saveTaskListData(taskList);
+        assertTrue(areFilesSame("src/test/java/athena/loadTask.csv", "src/test/java/athena/TaskAnswer.csv"));
```
Remember to add these files to the .gitignore

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    private boolean areFilesSame(String file1, String file2) {
+        BufferedReader answer = null;
```
I think `expected` vs `actual` would be a more consistent naming since this was used for our IO tests. But this is fine too.

Other than that, since this is a `BufferedReader` object, it would be nice to have the variable name sound like it, for example `answerFileReader` or `expectedFileReader`? Same goes for `attempt` in this method.

daniellimws commented on src/test/java/athena/StorageTest.java:
```diff
@@ -0,0 +1,53 @@
+            while ((answerLine != null) || (attemptLine != null)) {
+                if (!answerLine.contentEquals(attemptLine)) {
+                    return false;
+                } else {
+                    answerLine = answer.readLine();
+                    attemptLine = attempt.readLine();
+                }
```
`else` is not needed since this will always be reached when the `if` condition is not satisfied.

daniellimws reviewed (changes requested):
Looks good. Left some comments about code quality in the test file. It would also be helpful to add Javadoc comments.

Also, shall our task numbers start from 1 instead of 0?

Kafcis committed:
Updated StorageTest

Kafcis committed:
Updated names of files

Kafcis committed:
Updated StorageTask

Kafcis committed:
Add testing files to .gitignore
